stdlib/asm/math/u64.masm

```diff
@@ -326,6 +326,65 @@ export.checked_eqz
     and
 end
 
+# Performs greater-than comparison of two unsigned 64 bit integers and returns min value.
+# The input values are assumed to be represented using 32 bit limbs, but this is not checked.
+# Stack transition looks as follows:
+# [b_hi, b_lo, a_hi, a_lo, ...] -> [c, ...], where c = a when a < b, and b otherwise.
+export.unchecked_min
+    dup.3
+    dup.3
+    dup.3
+    dup.3
+    exec.unchecked_gt
+    dup.0
+    movup.4
+    movup.3
+    movup.2
+    cdrop
+    movdn.3
+    cdrop
+    swap
+end
+
+# Performs greater-than comparison of two unsigned 64 bit integers and returns min value.
+# The input values are assumed to be represented using 32 bit limbs, fails if they are not.
+# Stack transition looks as follows:
+# [b_hi, b_lo, a_hi, a_lo, ...] -> [c, ...], where c = a when a < b, and b otherwise.
+export.checked_min
+    exec.u32assert4
+    exec.unchecked_min
+end
+
+# Performs less-than comparison of two unsigned 64 bit integers and returns max value.
+# The input values are assumed to be represented using 32 bit limbs, but this is not checked.
+# Stack transition looks as follows:
+# [b_hi, b_lo, a_hi, a_lo, ...] -> [c, ...], where c = a when a > b, and b otherwise.
+export.unchecked_max
+    dup.3
+    dup.3
+    dup.3
+    dup.3
+    exec.unchecked_lt
+    dup.0
+    movup.4
+    movup.3
+    movup.2
+    cdrop
+    movdn.3
+    cdrop
+    swap
```

Comment:
You can probably save a couple of cycles by using the approach mentioned in the other comment.